how-to-release.md

```diff
@@ -0,0 +1,298 @@
+
+If there are multiple volunteers, the steering committee should discuss and select a release manager, preferrably on a first-come-first-serve basis. All the other volunteers will be backup candidates if the designated one is unavailable or unable to perform their duties.
+
+If there is no volunteer or backup, the steering committee will need to discuss and choose one from the previous release managers as the release manager.
+
+Release managers should take responsibility to track the release progress and perform the actions during the whole release phase, and is responsible for all releases associated with a major release(i.e., 0.7.x, which included 0.7.0 through 0.7.3).
+
+# Prepare for the release
+
+Before your first release, you should perform one-time configuration steps. This will set up your security keys for signing the release and access to various release repositories.
+
+## One-time configuration steps
+
+1. A set of GPG signing key and password, which will be used for signing the artifacts. follow these [Github instructions](https://help.github.com/en/articles/generating-a-new-gpg-key) for generating and configuring one.
+
+GnuPG 2.1 switched to a new keyring format, for which the keys are stored in the `pubring.kbx` file. Since the release requires a `secring.gpg` file, you could create one like this:
+
+```
+# List the keys
+gpg --list-secret-keys --keyid-format LONG
+
+# Locate the key id, which is of the form `Dg4D421F...`
+
+# Create the secring.gpg
+gpg --export-secret-key <Your key-id> > ~/.gnupg/secring.gpg
+```
+
+2. Sonatype username/password with access to the `io.pravega` group. See this [guide](http://central.sonatype.org/pages/ossrh-guide.html) for how to create an account. You need to comment in this [ticket](https://issues.sonatype.org/browse/OSSRH-30708) and ask someone who already has deployer access to approve in order to get the permission to the `io.pravega` group.
+
+3. Dockerhub username/password with access to the Pravega repository (Only required for Pravega)
+
+For setting up docker see: https://docs.docker.com/engine/install/ubuntu/
+
+For configuring to run as a non-root user see: https://docs.docker.com/engine/install/linux-postinstall/, https://docs.docker.com/engine/security/rootless/
+
+## Announcing the release process starts
+When starting the release process, Pravega release managers should create a new release specific channel on `pravega.io` slack with name like `pravega-release-090` and keep posting the progress on the release on this channel from time to time. All the ecosystem projects release managers should be invited into this channel for continuous discussion. All contributors can join this channel and discuss about the release in it, but still keep in mind that the candidate announcing and voting should still happen and only take effect on the dev mailing list.
```

Comment:
When starting the release process, a Pravega release manager creates a new release Slack channel on `pravega.io` named after the release version, e.g., `pravega-release-090` for release `0.9.0`. The release manager posts release progress as new steps are executed; reporting progress is critical so that the community knows what is happening. All the ecosystem projects release managers must be invited into this channel for continuous discussion. All contributors can join this channel and discuss about the release in it, but still keep in mind that the candidate announcing and voting should still happen and only take effect on the dev mailing list. To let the community know that cutting a release is in progress, it is important to announce it on the `#dev` Slack channel.